### ISSUE TITLE
False positive on switch expression with await 

### ISSUE THREAD

**Issue by karol-gro** [DotNetAnalyzers/StyleCopAnalyzers#3460]
Stylecop version: 1.2.0-beta.406

I get false positive issue on following code:
```csharp
// this gives SA1119 
await (someBool switch
{
	true => DoOneThing(),
	false => DoAnother(),
});

// after applying suggestions we got not compiling code (await is applied on bool, so parentheses are required to change order of operations)
await someBool switch
{
	true => DoOneThing(),
	false => DoAnother(),
};
```
Of course in this case one should use `if`, but this is minimized version of issue found in our code
